### ISSUE TITLE
Add a Flush() to the interface

### ISSUE THREAD

**Issue by matthewmueller** [apex/log#49]
This would require some updates, but it would help for handlers like papertrail and ES to have the ability to buffer, then flush. For ones that go to stdout, it could be a no-op.

**Comment by tj** [comment]
agreed! It could be optional as a separate interface too I suppose, or just `io.Flusher`, but the internal logger I use for Ping has a Flush() method haha so I'm down
